crates/db_schema/src/impls/local_user.rs

```diff
@@ -62,8 +62,51 @@ impl LocalUser {
       .get_result(conn)
       .await
   }
+
+  // TODO: maybe move this and pass in LocalUserView
+  pub async fn export_backup(
+    pool: &mut DbPool<'_>,
+    person_id_: PersonId,
+  ) -> Result<UserBackupLists, Error> {
+    use crate::schema::{community, community_block, community_follower, person, person_block};
+    let conn = &mut get_conn(pool).await?;
+
+    let followed_communities = community_follower::dsl::community_follower
+      .filter(community_follower::person_id.eq(person_id_))
+      .inner_join(community::table.on(community_follower::community_id.eq(community::id)))
+      .select(community::actor_id)
+      .get_results(conn)
+      .await?;
+
+    let blocked_communities = community_block::dsl::community_block
+      .filter(community_block::person_id.eq(person_id_))
+      .inner_join(community::table)
+      .select(community::actor_id)
+      .get_results(conn)
+      .await?;
+
+    let blocked_users = person_block::dsl::person_block
+      .filter(person_block::person_id.eq(person_id_))
+      .inner_join(person::table.on(person_block::target_id.eq(person::id)))
+      .select(person::actor_id)
+      .get_results(conn)
+      .await?;
+
+    // TODO: use join for parallel queries?
```

Comment:
Yep you should be able to use `left_join`s for these. Look at the `select(` statements in the db_views types for how to select data from multiple joins.

Reply:
I'll play around with this today.

Reply:
Nevermind, I realized that since its a ton of left joins, and these are one -> many based off the person table, there would be a ton of duped cells.